build/commands/lib/util.js

```diff
@@ -876,6 +876,12 @@ const util = {
     const excludeFileName = util.getGitInfoExcludeFileName(repoDir, true)
     fs.appendFileSync(excludeFileName, '\n' + exclusion)
   },
+
+  fetchAndCheckoutRef: (repoDir, ref) => {
+    const options = { cwd: repoDir, stdio: 'inherit' }
+    util.run('git', ['fetch', 'origin', ref.replace(/^origin\//, '')], options)
+    util.run('git', ['-c', 'advice.detachedHead=false', 'checkout', ref], options)
```

Comment:
maybe `s/ref/FETCH_HEAD`?
If ref is sha-1 it's not clear where git stores it locally.

Reply:
upd: this should be good for sha1, but for tags/branch this could be an issue.